### DISCUSSION TITLE
how to set data in state from  getInitialProps and UseEffect simultaneously

### DISCUSSION THREAD

**Discussion by fateme-ramezanpour** [vercel/next.js#21782] [Help]
Hi,
I use getInitialProps and UseEffect for dispatch action on the same page, after that, I store data from getInitialProps and UseEffect in the same reducer, but data from the server do not show in the state. according to next-redux-wrapper documents  
The server and Client state must be a separation, but it makes the app complicated.  
is there any solution to resolve this problem, I mean to not separate client and server? could you give me an example?

**Comment by vendramini** [comment]
Sorry but I didn't understand your problem. Are you using this library? https://github.com/kirill-konshin/next-redux-wrapper
If so, and if you followed all the steps, it should work perfectly. Maybe with some code we could help you.

Still, you can reach the same result using react's built-in context api. I recommend leaving redux and use something like this: https://dev.to/netlify/using-react-context-for-state-management-in-next-js-4oof
